crates/typst-library/src/meta/provide.rs

```diff
@@ -0,0 +1,43 @@
+use crate::prelude::*;
+
+/// Provide custom metadata for the Typst query system.
+///
+/// The Typst query system allows users to extract metadata from the document, offering both a
+/// generic selector and a specific key-value mechanism. The 'provide' command is an essential
+/// part of this mechanism, associating a single piece of metadata with the specified key.
+/// Subsequent invocations will append to a list of values for the same key.
+///
+/// Example:
+/// ```example
+/// #provide("note", "This is a note")
```

Comment:
maybe add a bit more structured objects to example, e.g. a dictionary to show how it serializes. if you do, please add linebreaks so that no line is longer than 38 chars. then, it will render nicely in the documentation.